### ISSUE TITLE
Changing image layouts in release/acquire image barriers triggers false positives.

### ISSUE THREAD

**Issue by Themaister** [KhronosGroup/Vulkan-LoaderAndValidationLayers#1986]
When using two queues with different families, it is possible to trigger a misleading error about image layouts.

E.g. in a TRANSFER queue
```
srcQueue = TRANSFER
dstQueue = GRAPHICS
oldLayout = TRANSFER_DST_OPTIMAL
newLayout = SHADER_READ_ONLY_OPTIMAL
srcAccess = TRANSFER_WRITE
dstAccess = 0 // semaphore takes care of it
```
this signals a semaphore which graphics waits on, and needs to do the same layout transition. The transition only actually happen once according to spec.

Graphics:
```
srcQueue = TRANSFER
dstQueue = GRAPHICS
oldLayout = TRANSFER_DST_OPTIMAL
newLayout = SHADER_READ_ONLY_OPTIMAL
srcAccess = 0 // semaphore takes care of it
dstAccess = SHADER_READ_BIT
```
The acquire barrier triggers an error where the command buffer expects TRANSFER_DST_OPTIMAL (because it's transitioning from it), but the transition actually hasn't happened yet.

```
[Vulkan]: Error: DS: Cannot submit cmd buffer using image (0xb6) [sub-resource: aspectMask 0x1 array layer 0, mip level 0], with layout VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL when first use is VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL.
```
The release operation probably should not change the layout for the purposes of the tracker.

**Comment by chrisforbes** [comment]
Looking into this as already investigating some other bad behavior in the same area.

**Comment by mark-lunarg** [comment]
Duplicate of #865? Will leave them both open for now since @chrisforbes bit.

**Comment by jzulauf-lunarg** [comment]
@chrisforbes -- is this still on your active TODO list?  I'm starting to look at #1550 and I don't want to duplicate efforts.

**Comment by chrisforbes** [comment]
@jzulauf-lunarg I don't have a lot of time to look at validation at the moment, but #2017 is pretty close to what you want for this.

**Comment by jzulauf-lunarg** [comment]
@chrisforbes  I'll pull and rebase from #2017, and move forward with a new PR.

**Comment by jzulauf-lunarg** [comment]
Should be fixed by 2521 above
